spec/controllers/collection_participants_controller_spec.rb

```diff
@@ -0,0 +1,442 @@
+        end
+      end
+
+      context "where the user is a maintainer" do
+        render_views
+        let(:current_role) { CollectionParticipant::MODERATOR }
+
+        context "where the collection has several participants" do
+          let!(:users) do
+            Array.new(3) do
+              user = FactoryGirl.create(:user)
+              FactoryGirl.create(
+                :collection_participant,
+                collection: collection,
+                pseud: user.default_pseud
+              )
+              user
+            end
+          end
+
+          it "displays the participants" do
+            get :index, params
+            expect(response).to have_http_status(:success)
+            users.each do |user|
+              expect(response.body).to include user.default_pseud.name
+            end
+          end
+        end
+      end
+    end
+  end
+
+  describe "update" do
+    let(:user) { FactoryGirl.create(:user) }
+    let(:collection) { FactoryGirl.create(:collection) }
+    let(:user_role) { CollectionParticipant::NONE }
+    let!(:user_participant) do
+      FactoryGirl.create(
+        :collection_participant,
+        pseud: user.default_pseud,
+        collection: collection,
+        participant_role: user_role
+      )
+    end
+    let(:id_to_update) { nil }
+    let(:params) do
+      {
+        collection_participant: {
+          participant_role:  CollectionParticipant::MEMBER,
+          id: id_to_update
+        }
+      }
+    end
+
+    before do
+      fake_login_known_user(user)
+    end
+
+    context "where there is no participant" do
+      it "displays an error and redirects to the index" do
+        put :update, params
+        it_redirects_to_with_error(root_path, "Which participant did you want to work with?")
+      end
+    end
+
+    context "where there is a participant" do
+      let(:participant) do
+        FactoryGirl.create(
+          :collection_participant,
+          collection: user_participant.collection,
+          participant_role: CollectionParticipant::NONE
+        )
+      end
+      let(:id_to_update) { participant.id }
+      context "where the user is not a collection maintainer" do
+        it "redirects to the collection page and displays an error" do
+          put :update, params
+          it_redirects_to_with_error(collection_path(collection), "Sorry, you're not allowed to do that.")
+        end
+      end
+
+      context "where the user is a collection maintainer" do
+        let(:user_role) { CollectionParticipant::MODERATOR }
+        context "where the participant is updated successfully" do
+          it "successfully updates and redirects to collection participants" do
+            put :update, params
+            it_redirects_to_with_notice(collection_participants_path(collection), "Updated #{participant.pseud.name}.")
+            expect(flash[:notice]).to eq "Updated #{participant.pseud.name}."
```

Comment:
I think this line is left over from before the change to the `it_redirects_to_with_notice`?